### ISSUE TITLE
recent leaks

### ISSUE THREAD

**Issue by lundman** [openzfsonosx/zfs#652]
Latest master when unloading SPL spit out
```
    0    kernel: (spl) SPL: stop spl_free_thread
    0    kernel: (spl) SPL: stop spl_free_thread, lock acquired, setting exit variable and waiting
    0    kernel: (spl) SPL: spl_free_thread_exit set to FALSE and exiting: cv_broadcasting
    0    kernel: (spl) SPL: spl_free_thread thread_exit
    0    kernel: (spl) SPL: spl_free_thread stop: while loop ended, dropping mutex
    0    kernel: (spl) SPL: spl_free_thread stop: destroying cv and mutex
    0    kernel: (spl) SPL: bsd_untimeout
    0    kernel: (spl) SPL: wait for taskqs to empty
    0    kernel: (spl) SPL: destroy taskq
    0    kernel: (spl) SPL: tsd unloading 0
    0    kernel: (spl) SPL: Warning: kmem_cache_destroy: 'kmem_alloc_160' (0xffffff80bc557870) not empty
    0    kernel: (spl) SPL: Warning: kmem_cache_destroy: 'kmem_va_4096' (0xffffff80bc553070) not empty
    0    kernel: (spl) SPL: Warning: kmem_cache_destroy: 'kmem_bufctl_cache' (0xffffff80bc552870) not empty
    0    kernel: (spl) SPL: Warning: kmem_cache_destroy: 'kmem_slab_cache' (0xffffff80bc552470) not empty
    0    kernel: (spl) SPL: Released 44 slabs
    0    kernel: (spl) SPL: vmem_destroy('kmem_default'): leaked 4096 bytes
    0    kernel: (spl) SPL: vmem_destroy('kmem_va'): leaked 32768 bytes
    0    kernel: (spl) SPL: vmem_destroy('kmem_msb'): leaked 40960 bytes
    0    kernel: (spl) SPL: vmem_destroy('kmem_metadata'): leaked 40960 bytes
    0    kernel: (spl) SPL: vmem_fini: stopped vmem_update.  Creating list and walking arenas.
    0    kernel: (spl)

    0    kernel: (spl) SPL: vmem_destroy('heap'): leaked 294912 bytes
    0    kernel: (spl) SPL: vmem_fini: walking spl_heap_arena, aka bucket_heap (pass 1)
    0    kernel: (spl) SPL: vmem_fini: calling vmem_xfree(spl_default_arena, ptr, 268435456);
    0    kernel: (spl) SPL: vmem_fini: walking spl_heap_arena, aka bucket_heap (pass 2)
    0    kernel: (spl) SPL: vmem_fini: walking bucket arenas...
    0    kernel: (spl) SPL: vmem_fini: walking vmem metadata-related arenas...
    0    kernel: (spl) SPL: vmem_fini walking the root arena (spl_default_arena)...
    0    kernel: (spl) SPL: vmem_fini destroying bucket heap
    0    kernel: (spl) SPL: vmem_destroy('bucket_heap'): leaked 294912 bytes
    0    kernel: (spl) SPL: vmem_fini destroying spl_bucket_arenas...
    0    kernel: (spl)  4096
    0    kernel: (spl)  8192
    0    kernel: (spl)  16384
    0    kernel: (spl)  32768
    0    kernel: (spl)  65536
    0    kernel: (spl)  131072
    0    kernel: (spl)  262144
    0    kernel: (spl)  524288
    0    kernel: (spl)  1048576
    0    kernel: (spl)  2097152
    0    kernel: (spl)  4194304
    0    kernel: (spl)  8388608
    0    kernel: (spl)  16777216
    0    kernel: (spl)
    0    kernel: (spl) SPL: vmem_fini destroying vmem_vmem_arena
    0    kernel: (spl) SPL: vmem_destroy('vmem_vmem'): leaked 26352 bytes
    0    kernel: (spl) SPL: vmem_destroy('vmem_vmem'): STILL 26352 bytes at kstat_delete() time
    0    kernel: (spl) SPL: vmem_fini destroying vmem_seg_arena
    0    kernel: (spl) SPL: vmem_destroy('vmem_seg'): leaked 180224 bytes
    0    kernel: (spl) SPL: vmem_destroy('vmem_seg'): STILL 180224 bytes at kstat_delete() time
    0    kernel: (spl) SPL: vmem_fini destroying vmem_hash_arena
    0    kernel: (spl) SPL: vmem_fini destroying vmem_metadata_arena
    0    kernel: (spl) SPL: vmem_destroy('vmem_metadata'): leaked 180224 bytes
    0    kernel: (spl) SPL: vmem_destroy('vmem_metadata'): STILL 180224 bytes at kstat_delete() time
    0    kernel: (spl)

    0    kernel: (spl) SPL: vmem_destroy('spl_default_arena'): leaked 197271 bytes
    0    kernel: (spl) SPL: vmem_destroy('spl_default_arena'): STILL 197271 bytes at kstat_delete() time
    0    kernel: (spl) SPL: vmem_fini destroying spl_default_arena_parent
    0    kernel: (spl) SPL: arenas removed, now try destroying mutexes...
    0    kernel: (spl) vmem_xnu_alloc_lock
    0    kernel: (spl) vmem_panic_lock
    0    kernel: (spl) vmem_pushpage_lock
    0    kernel: (spl) vmem_nosleep_lock
    0    kernel: (spl) vmem_sleep_lock
    0    kernel: (spl) vmem_segfree_lock
    0    kernel: (spl) vmem_list_lock
    0    kernel: (spl)

    0    kernel: (spl) SPL: WOULD HAVE released 1273271 bytes (82 spans) from arenas
    0    kernel: (spl) SPL: vmem_fini: Brief delay for readability...
    0    kernel: (spl) SPL: vmem_fini: done!
    0    kernel: (spl) Dumping leaked mutex allocations...
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_mech_tabs.c kcf_init_mech_tabs 161 : # leaks: 1
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_mech_tabs.c kcf_init_mech_tabs 242 : # leaks: 276
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_prov_tabs.c kcf_prov_tab_init 82 : # leaks: 1
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_sched.c kcf_sched_init 1124 : # leaks: 1
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_sched.c kcf_sched_init 1134 : # leaks: 16
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_sched.c kcfpool_alloc 1288 : # leaks: 1
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_sched.c kcfpool_alloc 1291 : # leaks: 1
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_sched.c kcf_sched_init 1142 : # leaks: 1
    0    kernel: (spl)   mutex <private> : ../icp/core/kcf_prov_tabs.c kcf_alloc_provider_desc 350 : # leaks: 3
    0    kernel: (spl)   mutex <private> : aggsum.c aggsum_init 87 : # leaks: 7
    0    kernel: (spl)   mutex <private> : aggsum.c aggsum_init 93 : # leaks: 14
    0    kernel: (spl) Dumped 322 leaked allocations. Wait for watchdog to exit..
    0    kernel: (spl) SPL: watchdog thread exit
    0    kernel: (spl) SPL: Unloaded module v1.7.3-1-gf93f1008 (os_mem_alloc: 0)

```

**Comment by lundman** [comment]
_0    kernel: (spl)   mutex <private> : aggsum.c aggsum_init 87 : # leaks: 7_
_0    kernel: (spl)   mutex <private> : aggsum.c aggsum_init 93 : # leaks: 14_

https://github.com/openzfs/openzfs/pull/679

So in addition to missing `mutex_destroy()` we have some memory leak.

**Comment by lundman** [comment]
e027896406cad529324c06103ffc9657296c2261
952f9ba18ba572a9f07007a56b9ea1583b0c441d
97a49611045aea913cf6f88371339c5a75a5a718
1ba09e4b3f31260a41699b75847ae6f24e91c3cc

Now we finish with
<details>
  <summary> SPL unload </summary>
  <p>


```
    <spl`spl_kmem_thread_fini (spl-kmem.c:5406)> SPL: stop spl_free_thread
    <spl`spl_kmem_thread_fini (spl-kmem.c:5407)> SPL: stop spl_free_thread, lock acquired, setting exit variable and waiting
    <spl`spl_free_thread (spl-kmem.c:4966)> SPL: spl_free_thread_exit set to FALSE and exiting: cv_broadcasting
    <spl`spl_free_thread (spl-kmem.c:4970)> SPL: spl_free_thread thread_exit
    <spl`spl_kmem_thread_fini (spl-kmem.c:5414)> SPL: spl_free_thread stop: while loop ended, dropping mutex
    <spl`spl_kmem_thread_fini (spl-kmem.c:5416)> SPL: spl_free_thread stop: destroying cv and mutex
    <spl`spl_kmem_thread_fini (spl-kmem.c:5420)> SPL: bsd_untimeout
    <spl`spl_kmem_thread_fini (spl-kmem.c:5425)> SPL: wait for taskqs to empty
    <spl`spl_kmem_thread_fini (spl-kmem.c:5428)> SPL: destroy taskq
    <spl`spl_tsd_fini (spl-tsd.c:373)> SPL: tsd unloading 0
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_bufctl_cache' (0xffffff889002a870) not empty
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_slab_cache' (0xffffff889002a470) not empty
    <spl`spl_kmem_fini (spl-kmem.c:4307)> SPL: Released 182 slabs
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('kmem_msb'): leaked 311296 bytes
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('kmem_metadata'): leaked 311296 bytes
    <spl`vmem_fini (spl-vmem.c:3474)> SPL: vmem_fini: stopped vmem_update.  Creating list and walking arenas.
    <spl`vmem_fini (spl-vmem.c:3499)>

    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('heap'): leaked 1212416 bytes
    <spl`vmem_fini (spl-vmem.c:3503)> SPL: vmem_fini: walking spl_heap_arena, aka bucket_heap (pass 1)
    <spl`vmem_fini (spl-vmem.c:3514)> SPL: vmem_fini: calling vmem_xfree(spl_default_arena, ptr, 134217728);
    <spl`vmem_fini (spl-vmem.c:3519)> SPL: vmem_fini: walking spl_heap_arena, aka bucket_heap (pass 2)
    <spl`vmem_fini (spl-vmem.c:3521)> SPL: vmem_fini: walking bucket arenas...
    <spl`vmem_fini (spl-vmem.c:3534)> SPL: vmem_fini: walking vmem metadata-related arenas...
    <spl`vmem_fini (spl-vmem.c:3554)> SPL: vmem_fini walking the root arena (spl_default_arena)...
    <spl`vmem_fini (spl-vmem.c:3558)> SPL: vmem_fini destroying bucket heap
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('bucket_heap'): leaked 1212416 bytes
    <spl`vmem_fini (spl-vmem.c:3562)> SPL: vmem_fini destroying spl_bucket_arenas...
    <spl`vmem_fini (spl-vmem.c:3564)>  4096
    <spl`vmem_fini (spl-vmem.c:3564)>  8192
    <spl`vmem_fini (spl-vmem.c:3564)>  16384
    <spl`vmem_fini (spl-vmem.c:3564)>  32768
    <spl`vmem_fini (spl-vmem.c:3564)>  65536
    <spl`vmem_fini (spl-vmem.c:3564)>  131072
    <spl`vmem_fini (spl-vmem.c:3564)>  262144
    <spl`vmem_fini (spl-vmem.c:3564)>  524288
    <spl`vmem_fini (spl-vmem.c:3564)>  1048576
    <spl`vmem_fini (spl-vmem.c:3564)>  2097152
    <spl`vmem_fini (spl-vmem.c:3564)>  4194304
    <spl`vmem_fini (spl-vmem.c:3564)>  8388608
    <spl`vmem_fini (spl-vmem.c:3564)>  16777216
    <spl`vmem_fini (spl-vmem.c:3571)>
    <spl`vmem_fini (spl-vmem.c:3572)> SPL: vmem_fini destroying vmem_vmem_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_vmem'): leaked 26352 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_vmem'): STILL 26352 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3577)> SPL: vmem_fini destroying vmem_seg_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_seg'): leaked 753664 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_seg'): STILL 753664 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3582)> SPL: vmem_fini destroying vmem_hash_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_hash'): leaked 512 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_hash'): STILL 512 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3591)> SPL: vmem_fini destroying vmem_metadata_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_metadata'): leaked 757760 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_metadata'): STILL 757760 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3594)>

    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('spl_default_arena'): leaked 787095 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('spl_default_arena'): STILL 787095 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3597)> SPL: vmem_fini destroying spl_default_arena_parent
    <spl`vmem_fini (spl-vmem.c:3601)> SPL: arenas removed, now try destroying mutexes...
    <spl`vmem_fini (spl-vmem.c:3602)> vmem_xnu_alloc_lock
    <spl`vmem_fini (spl-vmem.c:3604)> vmem_panic_lock
    <spl`vmem_fini (spl-vmem.c:3606)> vmem_pushpage_lock
    <spl`vmem_fini (spl-vmem.c:3608)> vmem_nosleep_lock
    <spl`vmem_fini (spl-vmem.c:3610)> vmem_sleep_lock
    <spl`vmem_fini (spl-vmem.c:3612)> vmem_segfree_lock
    <spl`vmem_fini (spl-vmem.c:3614)> vmem_list_lock
    <spl`vmem_fini (spl-vmem.c:3626)>

    <spl`vmem_fini (spl-vmem.c:3636)> SPL: WOULD HAVE released 4818359 bytes (228 spans) from arenas
    <spl`vmem_fini (spl-vmem.c:3638)> SPL: vmem_fini: Brief delay for readability...
    <spl`vmem_fini (spl-vmem.c:3640)> SPL: vmem_fini: done!
    <spl`spl_mutex_subsystem_fini (spl-mutex.c:151)> Dumping leaked mutex allocations...
    <spl`spl_mutex_subsystem_fini (spl-mutex.c:198)> Dumped 0 leaked allocations. Wait for watchdog to exit..
    <spl`spl_wdlist_check (spl-mutex.c:105)> SPL: watchdog thread exit
    <spl`spl_stop (spl-osx.c:484)> SPL: Unloaded module v1.7.3-1-g5f47e5e (os_mem_alloc: 0)

```
</details>

**Comment by lundman** [comment]
After a full tester run:

<details>
  <summary> kext unload </summary>
  <p>


```
# kextunload -b net.lundman.zfs
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'zio_cache' (0xffffff889005f380) not empty
    <zfs`zfs_ioctl_osx_fini (zfs_ioctl.c:6923)> ZFS: Unloaded module v1.7.3-76-ge027896 (DEBUG mode)

# kextunload -b net.lundman.spl
    <spl`spl_kmem_thread_fini (spl-kmem.c:5228)> SPL: stop spl_free_thread
    <spl`spl_kmem_thread_fini (spl-kmem.c:5229)> SPL: stop spl_free_thread, lock acquired, setting exit variable and waiting
    <spl`spl_free_thread (spl-kmem.c:4821)> SPL: spl_free_thread_exit set to FALSE and exiting: cv_broadcasting
    <spl`spl_free_thread (spl-kmem.c:4825)> SPL: spl_free_thread thread_exit
    <spl`spl_kmem_thread_fini (spl-kmem.c:5236)> SPL: spl_free_thread stop: while loop ended, dropping mutex
    <spl`spl_kmem_thread_fini (spl-kmem.c:5238)> SPL: spl_free_thread stop: destroying cv and mutex
    <spl`spl_kmem_thread_fini (spl-kmem.c:5242)> SPL: bsd_untimeout
    <spl`spl_kmem_thread_fini (spl-kmem.c:5247)> SPL: wait for taskqs to empty
    <spl`spl_kmem_thread_fini (spl-kmem.c:5250)> SPL: destroy taskq
    <spl`spl_stop (spl-osx.c:475)> spl-rwlock.c spl_rwlock_fini 250 : ASSERTION(zfs_active_rwlock == 0) failed
    <spl`spl_tsd_fini (spl-tsd.c:373)> SPL: tsd unloading 0
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_alloc_48' (0xffffff889002da80) not empty
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_alloc_64' (0xffffff889002e380) not empty
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_alloc_96' (0xffffff889002ea80) not empty
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_va_4096' (0xffffff889002b000) not empty
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_va_8192' (0xffffff889002b380) not empty
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_bufctl_cache' (0xffffff889002a700) not empty
    <spl`cmn_err (spl-err.c:0)> SPL: Warning: kmem_cache_destroy: 'kmem_slab_cache' (0xffffff889002a380) not empty
    <spl`spl_kmem_fini (spl-kmem.c:4162)> SPL: Released 411 slabs
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('kmem_default'): leaked 45056 bytes
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('kmem_va'): leaked 229376 bytes
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('kmem_msb'): leaked 2285568 bytes
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('kmem_metadata'): leaked 2285568 bytes
    <spl`vmem_fini (spl-vmem.c:3474)> SPL: vmem_fini: stopped vmem_update.  Creating list and walking arenas.
    <spl`vmem_fini (spl-vmem.c:3499)>

    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('heap'): leaked 13369344 bytes
    <spl`vmem_fini (spl-vmem.c:3503)> SPL: vmem_fini: walking spl_heap_arena, aka bucket_heap (pass 1)
    <spl`vmem_fini (spl-vmem.c:3514)> SPL: vmem_fini: calling vmem_xfree(spl_default_arena, ptr, 134217728);
    <spl`vmem_fini (spl-vmem.c:3519)> SPL: vmem_fini: walking spl_heap_arena, aka bucket_heap (pass 2)
    <spl`vmem_fini (spl-vmem.c:3521)> SPL: vmem_fini: walking bucket arenas...
    <spl`vmem_fini (spl-vmem.c:3534)> SPL: vmem_fini: walking vmem metadata-related arenas...
    <spl`vmem_fini (spl-vmem.c:3554)> SPL: vmem_fini walking the root arena (spl_default_arena)...
    <spl`vmem_fini (spl-vmem.c:3558)> SPL: vmem_fini destroying bucket heap
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('bucket_heap'): leaked 13369344 bytes
    <spl`vmem_fini (spl-vmem.c:3562)> SPL: vmem_fini destroying spl_bucket_arenas...
    <spl`vmem_fini (spl-vmem.c:3564)>  4096
    <spl`vmem_fini (spl-vmem.c:3564)>  8192
    <spl`vmem_fini (spl-vmem.c:3564)>  16384
    <spl`vmem_fini (spl-vmem.c:3564)>  32768
    <spl`vmem_destroy (spl-vmem.c:2108)> SPL: vmem_destroy('bucket_32768'): leaked 3506176 bytes
    <spl`vmem_fini (spl-vmem.c:3564)>  65536
    <spl`vmem_fini (spl-vmem.c:3564)>  131072
    <spl`vmem_fini (spl-vmem.c:3564)>  262144
    <spl`vmem_fini (spl-vmem.c:3564)>  524288
    <spl`vmem_fini (spl-vmem.c:3564)>  1048576
    <spl`vmem_fini (spl-vmem.c:3564)>  2097152
    <spl`vmem_fini (spl-vmem.c:3564)>  4194304
    <spl`vmem_fini (spl-vmem.c:3564)>  8388608
    <spl`vmem_fini (spl-vmem.c:3564)>  16777216
    <spl`vmem_fini (spl-vmem.c:3571)>
    <spl`vmem_fini (spl-vmem.c:3572)> SPL: vmem_fini destroying vmem_vmem_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_vmem'): leaked 26304 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_vmem'): STILL 26304 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3577)> SPL: vmem_fini destroying vmem_seg_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_seg'): leaked 11714560 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_seg'): STILL 11714560 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3582)> SPL: vmem_fini destroying vmem_hash_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_hash'): leaked 6144 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_hash'): STILL 6144 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3591)> SPL: vmem_fini destroying vmem_metadata_arena
    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('vmem_metadata'): leaked 11722752 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('vmem_metadata'): STILL 11722752 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3594)>

    <spl`vmem_destroy_internal (spl-vmem.c:2157)> SPL: vmem_destroy('spl_default_arena'): leaked 11829911 bytes
    <spl`vmem_destroy_internal (spl-vmem.c:2196)> SPL: vmem_destroy('spl_default_arena'): STILL 11829911 bytes at kstat_delete() time
    <spl`vmem_fini (spl-vmem.c:3597)> SPL: vmem_fini destroying spl_default_arena_parent
    <spl`vmem_fini (spl-vmem.c:3601)> SPL: arenas removed, now try destroying mutexes...
    <spl`vmem_fini (spl-vmem.c:3602)> vmem_xnu_alloc_lock
    <spl`vmem_fini (spl-vmem.c:3604)> vmem_panic_lock
    <spl`vmem_fini (spl-vmem.c:3606)> vmem_pushpage_lock
    <spl`vmem_fini (spl-vmem.c:3608)> vmem_nosleep_lock
    <spl`vmem_fini (spl-vmem.c:3610)> vmem_sleep_lock
    <spl`vmem_fini (spl-vmem.c:3612)> vmem_segfree_lock
    <spl`vmem_fini (spl-vmem.c:3614)> vmem_list_lock
    <spl`vmem_fini (spl-vmem.c:3626)>

    <spl`vmem_fini (spl-vmem.c:3636)> SPL: WOULD HAVE released 65844631 bytes (2760 spans) from arenas
    <spl`vmem_fini (spl-vmem.c:3638)> SPL: vmem_fini: Brief delay for readability...
    <spl`vmem_fini (spl-vmem.c:3640)> SPL: vmem_fini: done!
    <spl`spl_stop (spl-osx.c:484)> SPL: Unloaded module v1.7.3-1-g5f47e5e (os_mem_alloc: 14680064)
```
</details>

**Comment by lundman** [comment]
These have been address - in particular the ARC abd miscounting.

If new leaks popup, create a new ticket.
